daemon/regmaild/regmaild.c

```diff
@@ -526,34 +548,32 @@ client_cb(int fd, short event, void *arg)
          req.cb != sizeof(req))
     {
         fprintf(stderr, "error: corrupted request.\n");
-        close(fd);
-        free(ev);
-        return;
+        goto end;
     }
 
-
     switch(req.operation)
     {
         case REGMAILDB_REQ_COUNT:
-            if (!*req.userid || !*req.email) {
-                err_request(&req, fd, ev);
-                return;
-            }
-            ret = regmaildb_check_email(req.email, strlen(req.email), req.userid);
+        {
+            if (!validate_regmaildb_request(&req))
+                goto end;
```

Comment:
I believe you can move this outside the switch / case.
After that you can either drop the default case, or mark it as not reached.

Reply:
Um, in later commit I added `validate_verifydb_request` and I would have to move them back. I'd like to just keep them like this.